src/cmds/service.c

```diff
@@ -11,15 +11,69 @@
 #include <string.h>
 #include <unistd.h>
 #include <stdlib.h>
+#include <mem/misc/pool.h>
+#include <util/dlist.h>
+#include <sys/types.h>
+#include <signal.h>
 
-static void srv_print_services(void) {
-	printf("PRINT LIST OF SERVICES IS NOT SUPPORTED YET\n");
+#define len 32
+
+struct service_list {
+	struct dlist_head dlist_item;
+	char service_name[len];
+	pid_t pid;
+};
+
+POOL_DEF(pool, struct service_list, sizeof(struct service_list));		 
+
+static DLIST_DEFINE(head);
+
+static void add_element_in_list(const char *name, pid_t service_pid) {
+  	struct service_list *new_element;
+
+  	new_element = pool_alloc(&pool);
+   
+  	strncpy(new_element->service_name,name,len);
+
+	new_element->pid = service_pid;
+   
+  	dlist_init(&new_element->dlist_item);
+  
+  	dlist_add_next(&new_element->dlist_item, &head); 
+}
+
+static void delete_elem(const char *name) {
+	struct service_list *tmp;
+	
+	dlist_foreach_entry(tmp, &head, dlist_item) {
+		if (strcmp(tmp->service_name, name) == 0) {
+			kill(tmp->pid, SIGKILL);
```

Comment:
`kill` can return error code. For example, it can fail if target process is dead for some reason.